Home_work21/Home_work21.cpp

```diff
@@ -0,0 +1,88 @@
+﻿#include <iostream>
+#include <string>
+#include <stack>
+#include <set>
+#include <map>
+#include <algorithm>
+#include <stack>
+
+unsigned int uniqueWordsCount(const std::string& line) {
+    std::string word = line;
+    std::set <std::string> container;
+    while (!word.empty()) {
+        int p = word.find(' ');
+        if (p < 0) {
+            container.insert(word);
+            break;
+        }
+        else {
+            container.insert(word.substr(0, p));
+            word.erase(0, p + 1);
+        }
```

Comment:
You can extract code to function to prevent duplication

or you can read words like this:
```c++
std::istringstream iss(line);
std::string word;
while (iss >> word) {
//do something
}